File: server/logger.cpp
#include "logger.hpp"

std::string Logger::timestamp() {
    auto now = std::chrono::system_clock::now();
    auto time_t = std::chrono::system_clock::to_time_t(now);
    std::stringstream ss;
    ss << std::put_time(std::localtime(&time_t), "%H:%M:%S");
    return ss.str();
}

void Logger::info(const std::string& msg) {
    std::cout << CYAN "[" << timestamp() << "] [INFO]" RESET " " << msg << std::endl;
}

void Logger::success(const std::string& msg) {
    std::cout << GREEN "[" << timestamp() << "] [OK]" RESET " " << msg << std::endl;
}

void Logger::warning(const std::string& msg) {
    std::cout << YELLOW "[" << timestamp() << "] [WARN]" RESET " " << msg << std::endl;
}

void Logger::error(const std::string& msg) {
    std::cout << RED "[" << timestamp() << "] [ERROR]" RESET " " << msg << std::endl;
}

void Logger::alert(const std::string& msg) {
    std::cout << RED BOLD "[" << timestamp() << "] [ALERT]" RESET " " << msg << std::endl;
}

void Logger::sum_result(const std::string& msg) {
    std::cout << GREEN BOLD "[" << timestamp() << "] [SUM]" RESET " " << msg << std::endl;
}

File: client/logger.py
from datetime import datetime

# ANSI color codes
RESET = '\033[0m'
RED = '\033[31m'
GREEN = '\033[32m'
YELLOW = '\033[33m'
BLUE = '\033[34m'
CYAN = '\033[36m'
BOLD = '\033[1m'

class Logger:
    @staticmethod
    def timestamp():
        return datetime.now().strftime("%H:%M:%S")

    @staticmethod
    def info(msg):
        print(f"{CYAN}[{Logger.timestamp()}] [INFO]{RESET} {msg}")

    @staticmethod
    def success(msg):
        print(f"{GREEN}[{Logger.timestamp()}] [OK]{RESET} {msg}")

    @staticmethod
    def warning(msg):
        print(f"{YELLOW}[{Logger.timestamp()}] [WARN]{RESET} {msg}")

    @staticmethod
    def error(msg):
        print(f"{RED}[{Logger.timestamp()}] [ERROR]{RESET} {msg}")
        